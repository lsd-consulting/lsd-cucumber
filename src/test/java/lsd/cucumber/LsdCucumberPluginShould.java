package lsd.cucumber;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import static java.nio.charset.Charset.defaultCharset;
import static java.nio.file.Files.readString;
import static java.nio.file.Paths.get;

class LsdCucumberPluginShould {

    @Test
    void generateMenu() throws IOException {
        Approvals.verifyHtml(readString(get("build/reports/lsd/lsd-index.html"), defaultCharset()));
    }

    @Test
    void generateReportForBasicScenario() throws IOException {
        Approvals.verifyHtml(readString(get("build/reports/lsd/Scenario.html"), defaultCharset()));
    }

    @Test
    void generateReportForScenarioWithDataTable() throws IOException {
        Approvals.verifyHtml(readString(get("build/reports/lsd/Data_table.html"), defaultCharset()));
    }

    @Test
    void generateForScenarioOutline() throws IOException {
        Approvals.verifyHtml(readString(get("build/reports/lsd/Scenario_outline.html"), defaultCharset()));
    }
}
