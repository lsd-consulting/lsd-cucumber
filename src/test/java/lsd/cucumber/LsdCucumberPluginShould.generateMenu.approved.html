<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xmlns="http://www.w3.org/1999/html" xml:lang="en" lang="en">
<head>
    <title></title>
    <link rel="stylesheet" href="style.css">
</head>
<body>
<main role="main">
    <h1 class="logo">LSD</h1>
    <h1>Reports</h1>
    <div class="contents">
        <ul class="contents">
                        <li class="success"><a href="Data_table.html">data_table</a></li>
                        <li class="success"><a href="Scenario.html">scenario</a></li>
                        <li class="success"><a href="Scenario_outline.html">scenario_outline</a></li>
                    </ul>
    </div>
</main>
</body>
</html>
